tests/openai/test_openai_agent_autolog.py

```diff
@@ -0,0 +1,245 @@
+            context_variables=context_variables,
+        )
+
+    assert (
+        response.messages[-1]["content"]
+        == "Hello, James! Your account details have been successfully printed."
+    )
+    traces = get_traces()
+    assert len(traces) == 1
+    trace = traces[0]
+    assert trace.info.status == "OK"
+    spans = trace.data.spans
+    assert len(spans) == 6  # 1 root + 1 function call + 2 get_chat_completion + 2 Completions
+    assert spans[0].name == "run"
+    assert spans[0].inputs["agent"]["name"] == "Agent"
+    assert spans[0].inputs["messages"] == messages
+    assert (
+        spans[0].outputs["messages"][-1]["content"]
+        == "Hello, James! Your account details have been successfully printed."
+    )
+    assert spans[1].name == "Agent.get_chat_completion_1"
+    assert spans[1].parent_id == spans[0].span_id
+    assert spans[2].name == "Completions_1"
+    assert spans[2].parent_id == spans[1].span_id
+    assert spans[3].name == "Agent.print_account_details"
+    assert spans[3].span_type == SpanType.TOOL
+    assert spans[3].inputs["context_variables"] == context_variables
+    assert spans[3].outputs == "Success"
+    assert spans[3].parent_id == spans[0].span_id
+    assert spans[4].name == "Agent.get_chat_completion_2"
+    assert spans[4].parent_id == spans[0].span_id
+    assert spans[5].name == "Completions_2"
+    assert spans[5].parent_id == spans[4].span_id
+
+
+def test_autolog_swarm_agent_tool_exception(client):
+    mlflow.openai.autolog()
+
+    _DUMMY_RESPONSES = [_get_chat_completion(tool_call="always_fail")]
+
+    swarm = Swarm(client=client)
+
+    def always_fail():
+        raise Exception("This function always fails")
+
+    agent = Agent(name="Agent", instructions="You are an agent", functions=[always_fail])
+
+    with mock_openai(client, expected_responses=_DUMMY_RESPONSES):
+        messages = [{"role": "user", "content": "Hi!"}]
+        with pytest.raises(Exception, match="This function always fails"):
+            swarm.run(agent=agent, messages=messages)
+
+    traces = get_traces()
+    assert len(traces) == 1
+    trace = traces[0]
+    assert trace.info.status == "ERROR"
+    spans = trace.data.spans
+    assert len(spans) == 4  # 1 root + 1 function call + 1 get_chat_completion + 1 Completions
+    assert spans[3].span_type == SpanType.TOOL
+    assert spans[3].status.status_code == "ERROR"
+    assert spans[3].status.description == "Exception: This function always fails"
+    assert spans[3].events[0].name == "exception"
+
+
+def test_autolog_swarm_agent_completion_exception(client):
+    mlflow.openai.autolog()
+
+    swarm = Swarm(client=client)
+    agent = Agent(name="Agent", instructions="You are an agent")
+
+    with mock.patch.object(
+        client.chat.completions, "create", side_effect=RuntimeError("Connection error")
+    ):
+        messages = [{"role": "user", "content": "Hi!"}]
+        with pytest.raises(Exception, match="Connection error"):
```

Comment:
```suggestion
        with pytest.raises(RuntimeError, match="Connection error"):
```

Can we catch `RuntimeError`?